drivers/gles3/shader_gles3.cpp

```diff
@@ -564,6 +564,13 @@ bool ShaderGLES3::_load_from_cache(Version *p_version) {
 			specialization.id = glCreateProgram();
 			glProgramBinary(specialization.id, variant_format, variant_bytes.ptr(), variant_bytes.size());
 
+			GLint link_status = 0;
+			glGetProgramiv(specialization.id, GL_LINK_STATUS, &link_status);
+			if (link_status != GL_TRUE) {
+				WARN_PRINT("Couldn't load cached shader, recompiling.");
```

Comment:
Just tested (with the previous commit where it was still an error), I'd make it a `WARN_PRINT_ONCE` actually, as it's a bit spammy otherwise:
```
$ DRI_PRIME=0 godot-git
Godot Engine v4.1.dev.custom_build.4917101ae - https://godotengine.org
ERROR: Failed to load cached shader.
   at: _load_from_cache (drivers/gles3/shader_gles3.cpp:570)
ERROR: Failed to load cached shader.
   at: _load_from_cache (drivers/gles3/shader_gles3.cpp:570)
OpenGL API 4.6 (Core Profile) Mesa 23.1.0 - Compatibility - Using Device: Intel - Mesa Intel(R) HD Graphics 630 (KBL GT2)
 
ERROR: Failed to load cached shader.
   at: _load_from_cache (drivers/gles3/shader_gles3.cpp:570)
```

Reply:
done :)